SYCL/Basic/accessor/accessor.cpp

```diff
@@ -827,6 +878,73 @@ int main() {
     }
   }
 
+  // Test byte_size(), size(), max_size(), empty()
+  {
+    std::vector<char> vecChar(64, 'a');
+    std::vector<int> vecInt(32, 1);
+    std::vector<float> vecFloat(16, 1.0);
+    std::vector<IdxID3> vecCustom(8, {0, 0, 0});
+    TestAccSizeFuncs(vecChar);
+    TestAccSizeFuncs(vecInt);
+    TestAccSizeFuncs(vecFloat);
+    TestAccSizeFuncs(vecCustom);
+  }
+  // Test swap() on host_accessor
+  {
+    std::vector<int> vec1(8), vec2(16);
+    try {
+      sycl::buffer<int> buf1(vec1.data(), vec1.size());
+      sycl::buffer<int> buf2(vec2.data(), vec2.size());
+      auto acc1 = buf1.get_access<sycl::access::mode::read_write>();
+      auto acc2 = buf2.get_access<sycl::access::mode::read_write>();
+      acc1.swap(acc2);
+      acc1[15] = acc2[7] = 4;
+    } catch (sycl::exception &e) {
+      std::cout << e.what() << std::endl;
+    }
+    assert(vec1[7] == 4 && vec2[15] == 4);
+  }
+  // Test swap() on basic accessor
+  {
+    std::vector<int> vec1(8), vec2(16);
+    try {
+      sycl::buffer<int> buf1(vec1.data(), vec1.size());
+      sycl::buffer<int> buf2(vec2.data(), vec2.size());
+      sycl::queue q;
+      q.submit([&](sycl::handler &cgh) {
+        auto acc1 = buf1.get_access<sycl::access::mode::read_write>(cgh);
+        auto acc2 = buf2.get_access<sycl::access::mode::read_write>(cgh);
+        acc1.swap(acc2);
+        cgh.single_task([=]() { acc1[15] = acc2[7] = 4; });
+      });
+    } catch (sycl::exception &e) {
+      std::cout << e.what() << std::endl;
+    }
+    assert(vec1[7] == 4 && vec2[15] == 4);
+  }
+  // Test swap on local_accessor
+  {
+    std::vector<int> vec1(8), vec2(8);
+    try {
+      sycl::buffer<int> buf1(vec1.data(), vec1.size());
+      sycl::buffer<int> buf2(vec2.data(), vec2.size());
+
+      sycl::queue q;
+      q.submit([&](sycl::handler &cgh) {
+        sycl::local_accessor<int, 1> locAcc1(8, cgh), locAcc2(8, cgh);
+        // shouldn't fail
```

Comment:
I don't quite understand the need for this comment. Are you saying we are not actually verifying and just see that it doesn't crash? IMO, that's not enough.

Can we assign/swap/save onto normal buffers/device accessors and then assert on the host?

Reply:
I agree with you, but I have no idea how to properly test this feature on local_accessor. Maybe you can suggest any plan?

Reply:
I've just realized what I was missing. Can we perform `swap` on the device?

Reply:
That's it, we can't, because kernel accepts accessors by value, so we are unable to change their members. TBH I don't quite understand why does local_accessor need swap() method at all, looks like it's just a copy-paste for all accessors.

Reply:
Can we declare them with different sizes and then swap? E.g. one is 1-elem, another is N-elem. At least we'd ensure we won't crash.

Reply:
Yep, I'm also thinking of this solution, let's try.

Reply:
Done [9bd147a](https://github.com/intel/llvm-test-suite/pull/1265/commits/9bd147ad9b278fa38244aa52843b8d5f96120745)